detection-rules/shopify_infra_abuse.yml

```diff
@@ -0,0 +1,35 @@
+name: "Shopify infrastructure abuse"
+description: "Attackers have been observed using myshopify.com links to bypass domain reputation checks."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and any(body.links,
+          .href_url.domain.root_domain == "myshopify.com"
+  
+          // negate legitimate Shopify newsletters
+          and (
+            not any(headers.domains, .root_domain == "klaviyomail.com")
+            and not any(body.links,
+                        .href_url.domain.root_domain in ("klclick2.com")
+            )
+            and not any(headers.domains,
+                        .root_domain not in ("shopify.com", "myshopify.com")
```

Comment:
nested `not` here.  was this intentional? 

maybe each of these 3 conditions just needs to be it's own `and not ...`?